cdap-ui/app/cdap/components/AbstractWidget/InputKeyValueDropdown/InputKeyValueDropdown.tsx

```diff
@@ -0,0 +1,181 @@
+import * as React from 'react';
+
+import AbstractRow, {
+  AbstractRowStyles,
+  IAbstractRowProps,
+} from 'components/AbstractWidget/AbstractMultiRowWidget/AbstractRow';
+import { IStageSchema, IWidgetProps } from 'components/AbstractWidget';
+import { objectQuery } from 'services/helpers';
+import Input from '@material-ui/core/Input';
+import MenuItem from '@material-ui/core/MenuItem';
+import Select from '@material-ui/core/Select';
+import withStyles from '@material-ui/core/styles/withStyles';
+
+const styles = (theme) => {
+  return {
+    ...AbstractRowStyles(theme),
+    inputContainer: {
+      display: 'grid',
+      gridTemplateColumns: '50% 50%',
+      gridGap: '10px',
+    },
+    disabled: {
+      color: `${theme.palette.grey['50']}`,
+    },
+  };
+};
+
+interface IField {
+  name: string;
+  type: string[];
+}
+
+interface IComplexDropdown {
+  value: string | number;
+  label: string;
+}
+
+export type IDropdownOption = string | number | IComplexDropdown;
+
+interface IInputKeyValueDropdownProps extends IAbstractRowProps<typeof styles> {
+  keyPlaceholder?: string;
+  kvDelimiter?: string;
+  // dropdownOptions: IDropdownOption[];
+  dropdownOptions: any;
+  extraConfig?: any;
+}
+
+interface IKeyValueState {
+  value: string;
+  key: string;
+}
+
+type StateKeys = keyof IKeyValueState;
+
+class InputKeyValueDropdown extends AbstractRow<IInputKeyValueDropdownProps, IKeyValueState> {
+  public static defaultProps = {
+    keyPlaceholder: 'Test',
+    kvDelimiter: '=',
+    dropdownOptions: [],
+    delimiter: ',',
+  };
+
+  public state = {
+    key: '',
+    value: '',
+  };
+
+  public componentDidMount() {
+    const [key = '', value = ''] = this.props.value.split(this.props.kvDelimiter);
+
+    this.setState({
+      key,
+      value,
+    });
+  }
+
+  private handleChange = (type: StateKeys, e) => {
+    this.setState(
+      {
+        [type]: e.target.value,
+      } as Pick<IKeyValueState, StateKeys>,
+      () => {
+        const key = this.state.key;
+        const value = this.state.value;
+
+        const updatedValue = key.length > 0 ? [key, value].join(this.props.kvDelimiter) : '';
+        this.onChange(updatedValue);
+      }
+    );
+  };
+
+  public getFields(schemas: IStageSchema[], allowedTypes: string[]) {
```

Comment:
We seem to be calling `allowedTypes` always with `[]`. Is this really necessary?

Reply:
bump^